scripts/exgdas_enkf_sfc.sh

```diff
@@ -143,47 +138,65 @@ if [ $DOIAU = "YES" ]; then
+                COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL
+
+            MEMDIR=${memchar} RUN="enkfgdas" YMD=${gPDY} HH=${gcyc} generate_com \
+                COM_ATMOS_RESTART_MEM_PREV:COM_ATMOS_RESTART_TMPL
+
+            [[ ${TILE_NUM} -eq 1 ]] && mkdir -p "${COM_ATMOS_RESTART_MEM}"
 
-            $NCP $COMIN_GES_ENS/$memchar/atmos/RESTART/$bPDY.${bcyc}0000.sfc_data.tile${n}.nc $COMOUT_ENS/$memchar/atmos/RESTART/$bPDY.${bcyc}0000.sfcanl_data.tile${n}.nc
-            $NLN $COMIN_GES_ENS/$memchar/atmos/RESTART/$bPDY.${bcyc}0000.sfc_data.tile${n}.nc $DATA/fnbgsi.$cmem
-            $NLN $COMOUT_ENS/$memchar/atmos/RESTART/$bPDY.${bcyc}0000.sfcanl_data.tile${n}.nc $DATA/fnbgso.$cmem
-            $NLN $FIXfv3/$CASE/${CASE}_grid.tile${n}.nc                                $DATA/fngrid.$cmem
-            $NLN $FIXfv3/$CASE/${CASE}_oro_data.tile${n}.nc                            $DATA/fnorog.$cmem
+            ${NCP} "${COM_ATMOS_RESTART_MEM_PREV}/${bPDY}.${bcyc}0000.sfc_data.tile${n}.nc" \
+                "${COM_ATMOS_RESTART_MEM}/${bPDY}.${bcyc}0000.sfcanl_data.tile${n}.nc"
+            ${NLN} "${COM_ATMOS_RESTART_MEM_PREV}/${bPDY}.${bcyc}0000.sfc_data.tile${n}.nc" \
+                "${DATA}/fnbgsi.${cmem}"
+            ${NLN} "${COM_ATMOS_RESTART_MEM}/${bPDY}.${bcyc}0000.sfcanl_data.tile${n}.nc" \
+                "${DATA}/fnbgso.${cmem}"
+            ${NLN} "${FIXfv3}/${CASE}/${CASE}_grid.tile${n}.nc"     "${DATA}/fngrid.${cmem}"
+            ${NLN} "${FIXfv3}/${CASE}/${CASE}_oro_data.tile${n}.nc" "${DATA}/fnorog.${cmem}"
 
         done
 
-        $CYCLESH
+        CDATE="${PDY}${cyc}" ${CYCLESH}
         export err=$?; err_chk
 
     done
 
 fi
 
 if [ $DOSFCANL_ENKF = "YES" ]; then
- for n in $(seq 1 $ntiles); do
+    for n in $(seq 1 $ntiles); do
 
-    export TILE_NUM=$n
+        export TILE_NUM=$n
 
-    for imem in $(seq 1 $NMEM_ENKF); do
+        for imem in $(seq 1 $NMEM_ENKF); do
 
-        cmem=$(printf %03i $imem)
-        memchar="mem$cmem"
+            cmem=$(printf %03i $imem)
+            memchar="mem$cmem"
 
-        [[ $TILE_NUM -eq 1 ]] && mkdir -p $COMOUT_ENS/$memchar/atmos/RESTART
+            MEMDIR=${memchar} YMD=${PDY} HH=${cyc} generate_com \
+                COM_ATMOS_RESTART_MEM:COM_ATMOS_RESTART_TMPL
 
-        $NCP $COMIN_GES_ENS/$memchar/atmos/RESTART/$PDY.${cyc}0000.sfc_data.tile${n}.nc $COMOUT_ENS/$memchar/atmos/RESTART/$PDY.${cyc}0000.sfcanl_data.tile${n}.nc
-        $NLN $COMIN_GES_ENS/$memchar/atmos/RESTART/$PDY.${cyc}0000.sfc_data.tile${n}.nc $DATA/fnbgsi.$cmem
-        $NLN $COMOUT_ENS/$memchar/atmos/RESTART/$PDY.${cyc}0000.sfcanl_data.tile${n}.nc $DATA/fnbgso.$cmem
-        $NLN $FIXfv3/$CASE/${CASE}_grid.tile${n}.nc                               $DATA/fngrid.$cmem
-        $NLN $FIXfv3/$CASE/${CASE}_oro_data.tile${n}.nc                           $DATA/fnorog.$cmem
+            MEMDIR=${memchar} YMD=${gPDY} HH=${gcyc} generate_com \
+                COM_ATMOS_RESTART_MEM_PREV:COM_ATMOS_RESTART_TMPL
```

Comment:
For`DOSFCANL_ENKF`, `COM_ATMOS_RESTART_MEM_PREV` should be from `enkfgdas.PDY/cyc`, the line should be 
`MEMDIR=${memchar} YMD=${gPDY} HH=${gcyc} RUN=enkfgdas generate_com \
                 COM_ATMOS_RESTART_MEM_PREV:COM_ATMOS_RESTART_TMPL`

I tested on WCOSS2, and it succeeded after the modification

Reply:
Now setting `RUN` to `$GDUMP_enkf`, which is set in the j-job as 'enkfgdas'.

Reply:
> > 3\. Modify parm/config/config.resources from `if [[ "${_CDUMP}" =~ "gfs" ]]; then` to `if [[ "${_CDUMP}" = "gfs" ]]; then`
> 
> I had gone the other direction and extended the runtime for the enkfgfs forecast. But given how enkfgfs is going to be used (not run out long to also be the free forecast), this is probably correct. We can settle it in a future PR.

If I remembered correctly, for C384, the resources will not match to cause of the failure of the job. Yes, we can settle it in a future PR.